torch/nn/modules/batchnorm.py

```diff
@@ -684,9 +683,6 @@ def forward(self, input: Tensor) -> Tensor:
                 input, running_mean, running_var, self.weight, self.bias,
                 bn_training, exponential_average_factor, self.eps)
         else:
-            if not self.ddp_gpu_size:
```

Comment:
looks like it is intentionally restricting SyncBatchNorm to work with DDP only. feel it is not hard to keep it?

otherwise, we can completely kill _specify_ddp_gpu_num() and self.ddp_gpu_size

Reply:
I thought it was used for restricting DDP SPSD mode. Since now SPMD is gone, I am thinking of completely killing `_specify_ddp_gpu_num()` and `self.ddp_gpu_size`.

For now, `_specify_ddp_gpu_num()` is kept for BC only, and it only checks if the input module is a GPU module not CPU module.

Reply:
I see, I think it would be good to provide another way to check input module is a GPU module or not before completely retiring _specify_ddp_gpu_num() and self.ddp_gpu_size in the followup diffs. 

Thanks!

Reply:
The only 2 callers of `_specify_ddp_gpu_num`, both checked whether the module is a GPU module before invoking this function. Therefore, `_specify_ddp_gpu_num` becomes totally useless now. Will remove it in the next PR.